File: farewell.clj
(defn decrypt [encrypted]
  (let [key "i hope in the next ten years there would be no other farewell letter brilliant than this one"]
    (map bit-xor encrypted (map int (cycle key)))))

(let [content (slurp (first *command-line-args*))]
  (let [encrypted (map #(Integer/parseInt %) (re-seq #"\d+" (.replaceAll content "\\s", "")))]
    (print (apply str (map char (decrypt encrypted))))))

File: farewell.m
#!/usr/bin/env octave -q

key = int32('i hope in the next ten years there would be no other farewell letter brilliant than this one')';

fid = fopen(argv(){1});
content = strsplit(fscanf(fid, '%s'), ',');
[r, c] = size(content);
encrypted = zeros(c, 1);
for i = 1:c
  encrypted(i) = str2num(content{1, i});
endfor

printf('%s', char(bitxor(encrypted, repmat(key, ceil(c / size(key)(1)) + 1, 1)(1:c)))');
